core/commands/routing.go

```diff
@@ -301,6 +301,10 @@ var findPeerRoutingCmd = &cmds.Command{
 			return err
 		}
 
+		if pid == nd.Identity {
+			return ErrSelfUnsupported
```

Comment:
I guess this works but this seems very ad-hoc.
I think this is covering up a bug, ideally the routing layers like dht, ... would return this error.

This is better than nothing still.